communication/src/networking.rs

```diff
@@ -21,7 +21,7 @@ type ByteOrder = byteorder::BigEndian;
 /// Framing data for each `Vec<u8>` transmission, indicating a typed channel, the source and
 /// destination workers, and the length in bytes.
 // *Warning*: Adding, removing and altering fields requires to adjust the implementation below!
-#[derive(Abomonation, Debug, PartialEq, Eq, Hash, Clone, Copy)]
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy)]
```

Comment:
Adding in `, Serialize, Deserialize` here (one of the logging events has one of these, and can't derive without it). That might change how we want to do the reading and writing below, but happy to fix that up later.

Reply:
Added back in.